### PR TITLE
[static-build] add gatsby builder plugin injection

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Modifies the `@vercel/static-build` injection logic in order to now inject both the analytics and builder gatsby plugins.

### CODE REVIEW

File: packages/static-build/src/index.ts [hunk 1]
```diff
@@ -317,7 +322,11 @@ export const build: BuildV2 = async ({
 
   const pkg = getPkg(entrypoint, workPath);
   const devScript = pkg ? getScriptName(pkg, 'dev', config) : null;
-  const framework = getFramework(config, pkg);
+  const framework = await detectFrameworkRecord({
+    fs: new LocalFileSystemDetector(workPath),
+    frameworkList: frameworks,
+  });
+  // debug(JSON.stringify(f, null, 2));
```
Comment (TooTallNate) [inline packages/static-build/src/index.ts]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "@vercel/build-utils": "*",
     "@vercel/ncc": "0.24.0",
     "@vercel/next": "*",
+    "@vercel/static-build": "*",
```
Comment (styfle) [inline package.json:17]: Is this necessary?

Comment (Ethan-Arrowood) [reply package.json:17]: yes - same reason we had to put `@vercel/next` in here. Its used by the `test/lib/runBuildLambda.js`

Comment (Ethan-Arrowood) [reply package.json:17]: but i actually don't know if anything is actually running the tests in `packages/static-build/test/builds.test.js`

File: test/lib/deployment/now-deploy.js [hunk 3]
```diff
@@ -32,15 +32,21 @@ async function nowDeploy(projectName, bodies, randomness, uploadNowJson) {
 
   const { FORCE_BUILD_IN_REGION, VERCEL_DEBUG, VERCEL_CLI_VERSION } =
     process.env;
+  console.log({ VERCEL_CLI_VERSION });
```
Comment (Ethan-Arrowood) [inline test/lib/deployment/now-deploy.js]:
```suggestion
```

File: test/lib/deployment/now-deploy.js [hunk 4]
```diff
@@ -32,15 +32,21 @@ async function nowDeploy(projectName, bodies, randomness, uploadNowJson) {
...
   const nowJson = JSON.parse(
     bodies['vercel.json'] || bodies['now.json'] || '{}'
   );
+  console.log(nowJson);
```
Comment (Ethan-Arrowood) [inline test/lib/deployment/now-deploy.js]:
```suggestion
```

File: test/lib/deployment/now-deploy.js [hunk 5]
```diff
@@ -52,11 +58,9 @@ async function nowDeploy(projectName, bodies, randomness, uploadNowJson) {
         NEXT_TELEMETRY_DISABLED: '1',
       },
     },
-    name: projectName,
-    files,
-    meta: {},
-    ...nowJson,
   };
+  console.log(nowDeployPayload);
+  console.log(nowDeployPayload.build.env);
```
Comment (Ethan-Arrowood) [inline test/lib/deployment/now-deploy.js]:
```suggestion
```

File: packages/static-build/test/fixtures/gatsby-v4/gatsby-node.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    path: '/using-dsg',
+    component: require.resolve('./src/templates/using-dsg.js'),
+    context: {},
+    defer: true,
+  });
+};
+
+// exports.onPostBuild = async ({ store }) => {
+//   console.log(store.getState().pages);
+// };
```
Comment (Ethan-Arrowood) [inline packages/static-build/test/fixtures/gatsby-v4/gatsby-node.js]:
```suggestion
```

File: packages/static-build/test/fixtures/gatsby-v4/gatsby-node.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+// const fs = require('fs');
+
```
Comment (Ethan-Arrowood) [inline packages/static-build/test/fixtures/gatsby-v4/gatsby-node.js]:
```suggestion
```

File: package.json (hunk 2 above)
Comment (TooTallNate) [reply package.json:17]: In a separate PR we should refactor `runBuildLambda()` to accept an input `build()` function that would be imported from the builders' tests. That way these references at the root would not be necessary.

**Review by TooTallNate — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (TooTallNate) [reply package.json:17]: See https://github.com/vercel/vercel/pull/9262.

**Pull request merged**